gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/MysqlMultiActiveLeaseArbiterTestingDecorator.java

```diff
@@ -0,0 +1,206 @@
+package org.apache.gobblin.runtime.api;
+
+import com.google.common.base.Optional;
+import com.typesafe.config.Config;
+import java.io.IOException;
+import java.net.Inet6Address;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.HashMap;
+import javax.inject.Inject;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.gobblin.configuration.ConfigurationKeys;
+import org.apache.gobblin.util.ConfigUtils;
+import org.apache.gobblin.util.HostUtils;
+
+
+/**
+ * This class is a decorator for {@link MysqlMultiActiveLeaseArbiter} used to model scenarios where a lease owner fails
+ * to complete a lease intermittently (representing a variety of slowness or failure cases that can result on the
+ * participant side, network connectivity, or database).
+ *
+ * It will fail on calls to {@link MysqlMultiActiveLeaseArbiter.recordLeaseSuccess()} where a function of the lease
+ * obtained timestamp matches a bitmask of the host. Ideally, each participant should fail on different calls (with
+ * limited overlap if we want to test that). We use a deterministic method of failing some calls to complete a lease
+ * success with the following methodology. We take the binary representation of the lease obtained timestamp, scatter
+ * its bits through bit interleaving of the first and second halves of the binary representation to differentiate
+ * behavior of consecutive timestamps, and compare the last N digits (determined through config) to the bit mask of the
```

Comment:
I really like the idea of scattering (to avoid long runs of on/off).  while I have very little background in number theory, the idea of interleaving high-order with low-order bits intuitively concerns me... so not sure if I'm missing something.

here's my logic: when representing millis as a java long, we can essentially consider the high-order bits to be constants, since the lowest of the long's high bits [32:63] only changes once every 10^9 secs.  after interleaving, wouldn't odd position bits nearly never change, while even-positions would.  the assignment:
```
host0:0001,host1:0010,host2:0100,host3:1000
```
would give drastically different probability for `host1` and `host3`, than for `host0`, and `host2`--right?

instead I suggest bit "scattering" by passing the number through a hash digest, like MD5 or SHA1